### PR TITLE
Flexcheckout and order direct

### PR DESCRIPTION
**Pull request by robinmonjo**
Flexcheckout and order direct integration. The diff is huge because a lot of code was moved (to a super class) and because we added rubocop and made it auto-correct things ! But this is misleading, the gem is mostly the same with a `Ogone::Flexcheckout` and  `Ogone::OrderDirect`  class. 

<details>
No API were broken. 🏖
</details>

### CODE REVIEW

**Comment by robinmonjo** [comment]
@ssaunier please 🙏 I need you to run `gem owner ogone --add robin.monjo@applidget.com` otherwise I will have to publish the gem under another name 🙁...

**Review by imaireaux — APPROVED** [review]

**Review by Ahrry — APPROVED** [review]

File: lib/ogone/base.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    def ogone_host
+      @environment == 'test' ? 'ogone.test.v-psp.com' : 'secure.ogone.com'
+    end
+
+    private
+
+    def sha_in_sign
+      to_hash = sorted_upcased_parameters.each_with_object([]) do |(k, v), a|
+        a << "#{k}=#{v}#{@sha_in}" unless v.nil? || v == ''
```
Comment (Ahrry) [inline lib/ogone/base.rb]: `unless v.blank?`, I think.

File: lib/ogone/base.rb [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      end.join
+      sign to_hash
+    end
+
+    def sha_out_sign(params)
+      to_hash = self.class.const_get('OUTBOUND_SIGNATURE_PARAMETERS').each_with_object([]) do |p, a|
+        a << "#{p}=#{params[p]}#{@sha_out}" unless params[p].nil? || params[p] == ''
```
Comment (Ahrry) [inline lib/ogone/base.rb]: unless params[p].blank?, I think.

Comment (robinmonjo) [reply lib/ogone/base.rb]: blank is only available in rails not in pure ruby 😊

Comment (Ahrry) [reply lib/ogone/base.rb]: My bad !! 😅 👍

**Comment by ssaunier** [comment]
👌 

image

**Comment by robinmonjo** [comment]
Great !! Cheers @ssaunier 😊

**Pull request merged**